src/main/java/edu/phema/elm_to_omop/translate/PhemaElmToOmopTranslatorContext.java

```diff
@@ -4,130 +4,167 @@
 
-    public String getVocabularyReferenceForRetrieve(Retrieve retrieve) throws PhemaTranslationException {
-        Expression codeExpression = retrieve.getCodes();
-
-        if (codeExpression instanceof ValueSetRef) {
-            // The retrieve references a value set
-            return ((ValueSetRef) retrieve.getCodes()).getName();
-        } else if (codeExpression instanceof ToList) {
-            // The retrieve references a code
-            ToList toList = (ToList) codeExpression;
-            ToConcept toConcept = (ToConcept) toList.getOperand();
-            CodeRef codeRef = (CodeRef) toConcept.getOperand();
-
-            return codeRef.getName();
-        } else {
-            throw new PhemaTranslationException(String.format("Unable to retrieve codesetId for code expression of type %s", codeExpression.getClass().getSimpleName()));
-        }
+    Optional<PhemaConceptSet> result = conceptSets
+      .stream()
+      .filter(c -> c.getOid().equals(valueset.getId()))
+      .findFirst();
+
+    if (result.isPresent()) {
+      return result.get().id;
+    } else {
+      throw new PhemaTranslationException(String.format("Value set %s not found", valuesetReference));
     }
+  }
+
+  public int getCodeSetIdForVocabularyReference(String vocabularyReference) throws PhemaTranslationException {
+    String internalVocabularyReference = getInternalVocabularyReference(vocabularyReference);
 
-    public int getCodesetIdForRetrieve(Retrieve retrieve) throws PhemaTranslationException {
-        String vocabularyReference = getVocabularyReferenceForRetrieve(retrieve);
+    Optional<PhemaConceptSet> result = conceptSets
+      .stream()
+      .filter(c -> c.getOid().equals(internalVocabularyReference))
+      .findFirst();
 
-        return getCodeSetIdForVocabularyReference(vocabularyReference);
+    if (result.isPresent()) {
+      return result.get().id;
+    } else {
+      throw new PhemaTranslationException(String.format("Codeset not found for code %s", vocabularyReference));
     }
+  }
+
+  public String getVocabularyReferenceForRetrieve(Retrieve retrieve) throws PhemaTranslationException {
+    Expression codeExpression = retrieve.getCodes();
+
+    if (codeExpression instanceof ValueSetRef) {
+      // The retrieve references a value set
+      return ((ValueSetRef) retrieve.getCodes()).getName();
+    } else if (codeExpression instanceof ToList) {
+      // The retrieve references a code
+      ToList toList = (ToList) codeExpression;
+      Expression toListOperand = toList.getOperand();
+      CodeRef codeRef = null;
+      if (toListOperand instanceof ToConcept) {
+          ToConcept toConcept = (ToConcept) toList.getOperand();
+          codeRef = (CodeRef) toConcept.getOperand();
+      } else if (toListOperand instanceof CodeRef) {
+          codeRef = (CodeRef)toListOperand;
+      } else {
+          throw new PhemaTranslationException("Unsupported list operand");
+      }
```

Comment:
I added this quick fix in for some failing legacy (QUICK-based) tests.

Reply:
Oops, sorry. Thanks for fixing!